mfs/ops.go

```diff
@@ -3,10 +3,154 @@ package mfs
 import (
 	"errors"
 	"fmt"
+	"os"
+	gopath "path"
 	"strings"
+
+	dag "github.com/ipfs/go-ipfs/merkledag"
 )
 
-func rootLookup(r *Root, path string) (FSNode, error) {
+// Mv moves the file or directory at 'src' to 'dst'
+func Mv(r *Root, src, dst string) error {
+	srcDir, srcFname := gopath.Split(src)
+
+	var dstDirStr string
+	var filename string
+	if dst[len(dst)-1] == '/' {
+		dstDirStr = dst
+		filename = srcFname
+	} else {
+		dstDirStr, filename = gopath.Split(dst)
+	}
+
+	// get parent directories of both src and dest first
+	dstDir, err := lookupDir(r, dstDirStr)
+	if err != nil {
+		return err
+	}
+
+	srcDirObj, err := lookupDir(r, srcDir)
+	if err != nil {
+		return err
+	}
+
+	srcObj, err := srcDirObj.Child(srcFname)
+	if err != nil {
+		return err
+	}
+
+	nd, err := srcObj.GetNode()
+	if err != nil {
+		return err
+	}
+
+	fsn, err := dstDir.Child(filename)
+	if err == nil {
+		switch n := fsn.(type) {
+		case *File:
+			_ = dstDir.Unlink(filename)
+		case *Directory:
+			dstDir = n
+		default:
+			return fmt.Errorf("unexpected type at path: %s", dst)
+		}
+	} else if err != os.ErrNotExist {
```

Comment:
Its possible that it is, but just in case others are returned, i have this check here.